AppKit/CPView.j

```diff
@@ -1622,33 +1623,89 @@ var CPViewHighDPIDrawingEnabled = YES;
             while (view = [view superview]);
         }
 
-        [self _notifyViewDidHide];
+        [self _postViewWillDisappearNotification];
+        [self _recursiveGainedHiddenAncestor];
     }
     else
     {
         [self setNeedsDisplay:YES];
-        [self _notifyViewDidUnhide];
+
+        [self _postViewWillAppearNotification];
+        [self _recursiveLostHiddenAncestor];
     }
+
+    _isHidden = aFlag;
 }
 
-- (void)_notifyViewDidHide
+- (void)_postViewWillAppearNotification
 {
-    [self viewDidHide];
+    [[CPNotificationCenter defaultCenter] postNotificationName:_CPViewWillAppearNotification object:self userInfo:nil];
```

Comment:
It doesn't in cocoa. Makes sense to me because in -viewWillAppear, the subviews are already here. If you want to animate the subviews when the superview is added, it is still possible. You can do some setup (for example hide the subviews) and then start a animation in viewDidAppear.